lib/usagereporter/usagereporter.go

```diff
@@ -179,6 +186,9 @@ func (r *UsageReporter[T]) runSubmit(ctx context.Context) {
 			}
 
 			usageBatchSubmissionDuration.Observe(time.Since(t0).Seconds())
+		case <-r.gracefulStopEnqueued:
```

Comment:
Are we ok with only actually doing one final submission upon hitting a graceful stop? What about all the events that are potentially queued?

Reply:
That's a good question, IMO sending only one batch of events is ok. I do not think it is easy to send all events from the buffer without some major reactoring and removing the `runSubmit` loop 😬